### PR TITLE
New package: py-networkit (6.1, 7.0)

### PR DESCRIPTION
**Pull request by fabratu**
A follow-up to #14113. This is the Python-part, while the underlying CPP-lib is already part of spack #15714. The previous suggestions are part of this package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-networkit/package.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    # Required dependencies
+    depends_on('cmake', type='build')
+    depends_on('libnetworkit@7.0', type=('build', 'link'), when='@7.0')
+    depends_on('libnetworkit@6.1', type=('build', 'link'), when='@6.1')
+    depends_on('ninja', type='build')
+    depends_on('py-cython', type='build')
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-scipy', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-wheel', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-networkit/package.py]: Is wheel actually needed?

File: var/spack/repos/builtin/packages/py-networkit/package.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    depends_on('python@3.5:3.8', type=('build', 'run'))
+
+    phases = ['build_ext', 'install']
+
+    def build_ext_args(self, spec, prefix):
+        rpath = self.rpath
+        rpath.append(spec['libnetworkit'].prefix.lib)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-networkit/package.py]: Is this directory not already in `self.rpath`?

**Review by adamjstewart — COMMENTED** [review]
It looks like there's a way to build in parallel using the `--jobs` arg or `NETWORKIT_PARALLEL_JOBS` environment variable? We should use that to speed up the build.

I think a dependency on `llvm-openmp` is needed when building with Apple Clang. @alalazo is currently working on a PR to split `%clang` and `%apple-clang` into separate compilers. Once that's done, you can add:
```python
depends_on('llvm-openmp', when='%apple-clang')
```

File: var/spack/repos/builtin/packages/py-networkit/package.py (hunk 2 above)
Comment (fabratu) [reply var/spack/repos/builtin/packages/py-networkit/package.py]: No, it is not part of the rpath, but directly shared via `LD_LIBRARY_PATH`. When I started the first version of the package some time ago, this was not the case actually. The same goes for the dependency of `wheel`. Both are now deleted.

**Comment by fabratu** [comment]
Thanks for the input. I adjusted the package accordingly and added parallel building. If I understand the `llvm-openmp` part correctly, the integration/adaptation has to wait for the other PR to finish?

**Comment by alalazo** [comment]
> Thanks for the input. I adjusted the package accordingly and added parallel building. If I understand the llvm-openmp part correctly, the integration/adaptation has to wait for the other PR to finish?

@fabratu Feel free to go ahead with that. #17110 has been merged yesterday.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**